src/main/scala/com/johnsnowlabs/nlp/HasMultipleInputAnnotationCols.scala

```diff
@@ -0,0 +1,38 @@
+/*
+ * Copyright 2017-2021 John Snow Labs
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.johnsnowlabs.nlp
+
+/**
+ * Trait  used to create annotators with input columns of variable length.
+ * */
+trait HasMultipleInputAnnotationCols extends HasInputAnnotationCols {
```

Comment:
@maziyarpanahi @xusliebana , don't you guys think that the name for this trait could be `HasVariableCountInputAnnotationCols`

HasInputAnnotationCols already supports multiple input columns, the main change I see here is that the number of columns will no longer be defined in a 'declarative' fashion inside the library, but in a more dynamic way by the user, during annotator setup.

Reply:
@albertoandreottiATgmail Yes, that totally makes sense! I'll leave it to you and @xusliebana to rename this and the unit tests. (@xusliebana you can directly fix the changes inside release/333-release-candidate branch)

@albertoandreottiATgmail Could you please also review removing `final` from `final def setInputCols()`? I am not sure why we used `final` so I want to be sure removing it won't be risky in some scenarios.